### PR TITLE
add "Deny All" Dns Resolver

### PR DESCRIPTION
**Pull request by parkma99**
implement #329

### CODE REVIEW

File: rama-dns/src/deny_all.rs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+use std::fmt::Error;
+
+use crate::DnsResolver;
+
+#[derive(Debug, Clone)]
+pub struct DenyAllDns {}
```
Comment (GlenDC) [inline rama-dns/src/deny_all.rs:6]:
```suggestion
#[non_exhaustive]
/// a [`DnsResolver`] implementation which
/// denies all incoming DNS requests with a [`DnsDeniedError`].
pub struct DenyAllDns;

impl DenyAllDns {
    #[inline]
    /// Create a new [`Default`] [`DenyAllDns`].
    pub fn new() -> Self {
        Self::default()
    }
}

```

File: rama-dns/src/deny_all.rs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+use std::fmt::Error;
+
+use crate::DnsResolver;
+
+#[derive(Debug, Clone)]
```
Comment (GlenDC) [inline rama-dns/src/deny_all.rs:5]:
```suggestion
#[derive(Debug, Clone, Default)]
```

**Review by GlenDC — CHANGES_REQUESTED** [review]
Thanks, looks good.
Can you apply these changes,
and pull it onto your machine.

You might need to run `just qa` to quickly check if all passes.
It will also auto format in case there is some formatting to be done.

Should you not have `just` on your machine you probably can find install instructions
on <https://just.systems/man/en/installation.html> or for this instance you might get away to just check if `cargo fmt --all` doesn't give you new changes to commit.

If not too much to ask you can also squash your commits, but if not, also not ok, for now I'm still okay either way, it's what you want.

**Comment by parkma99** [comment]
Thanks, now all tests is passed, I find git is little confusing😊

**Comment by GlenDC** [comment]
> Thanks, now all tests is passed, I find git is little confusing😊

Yeah it is not something that one can just use without learning the nuances sadly.
Should you be interested I find <https://git-scm.com/book/en/v2> still the leading resource
that is worth reading at least once in your career.  Certainly most chapters, but not all.

Thanks for this contribution by the way, greatly appreciated.
Going to give it a last review, if it's okay and also all CI checks pass, I'll merge it in.
Otherwise I'll come back with more feedback.

Feel free to pick up another one, give feedback in any other way, or just leave it as-is.
Either way, have a nice life ahead of you and I wish you all the best!

**Review by GlenDC — APPROVED** [review]

**Pull request merged**